std/permutation.asm

```diff
@@ -0,0 +1,39 @@
+use std::prover::challenge;
+use std::array::fold;
+use std::array::len;
+use std::convert::int;
+use std::check::assert;
+
+let is_first: col = |i| if i == 0 { 1 } else { 0 };
+
+let alpha: expr = challenge(0, 1);
+let beta: expr = challenge(0, 2);
+
+// Maps [x_1, x_2, ..., x_n] to alpha**(n - 1) * x_1 + alpha ** (n - 2) * x_2 + ... + x_n
+let compress_expressions = |tuple| fold(tuple, 0, |acc, el| alpha * acc + el);
```

Comment:
Note that I did not include an ID here, like Chris did. Adding an ID allows us to share the accumulator column across  multiple permutations, at the cost of increasing the degree of the constraint.

Reply:
The tuple is an array.

Reply:
Maybe `compress_expression_array`?

Reply:
> what's `n` here? Not sure I understand this, where's `alpha**(n-1)` happening?

the fold computes `alpha * (alpha * ((alpha * 0) + x_1) + x_2) + x_3 ...`. If you expand that expression, you get the above.